src/content/zh-CN/2022/webassembly.md

```diff
@@ -0,0 +1,160 @@
+---
+#See https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters
+title: WebAssembly
+description: 2022 年 Web 年鉴的 WebAssembly 章节涵盖其用法、语言和 post-MVP 特性。
+authors: [ColinEberhardt]
+reviewers: [binji, RReverser]
+analysts: [JamieWhitMac]
+editors: [tunetheweb]
+translators: [luckybai]
+ColinEberhardt_bio: Colin 是 <a hreflang="en" href="https://www.scottlogic.com/">Scott Logic</a> 的 CTO，同时也是一个多产的技术作家、博客写作者以及一系列技术演讲人。他是 <a hreflang="en" href="https://www.finos.org/">FINOS</a> 的董事会成员, 这鼓励了金融部门的开源合作。他还在 GitHub 上非常活跃，为许多不同的项目做出了贡献。
+results: https://docs.google.com/spreadsheets/d/11jyABro0fKtuN6INnYP9pJlv5QWwp0jfJyTsGfKgScg/
+featured_quote: 尽管 WebAssembly 是一项小众技术，但它已经为 Web 带来了价值，有许多 Web 应用程序从这项技术中受益匪浅。
+featured_stat_1: 383
+featured_stat_label_1: 发现的独有的 WebAssembly 二进制文件
+featured_stat_2: 34.9%
+featured_stat_label_2: 使用 Amazon IVS 的移动端网站
+featured_stat_3: 72.8%
+featured_stat_label_3: 使用 Emscripten 脚本的模块
+---
+
+## 简介
+
+WebAssembly（简写 Wasm）是 Web 技术家族（JavaScript、HTML、CSS）中相对较新的成员，于 2019 年 12 月成为 W3C 官方认可的标准。
+
+WebAssembly 在浏览器中引入了一个新的运行时，该运行时与 JavaScript 运行时协同工作。相比之下，它更轻量，拥有少量的指令集以及严格的隔离模型（WebAssembly 默认没有 I/O）。开发 WebAssembly 的主要动机之一是为更多的编程语言（C++、Rust、Go 等）提供编译目标，允许开发人员使用更广泛的工具集开发新的 Web 应用程序或移植现有的应用程序。
+
+WebAssembly 的高光示例包括它<a hreflang="en" href="https://blog.chromium.org/2019/06/webassembly-brings-google-earth-to-more.html">在 Google Earth 中的使用</a>，它的 C++ 桌面应用程序现在可以在浏览器中使用。基于浏览器的设计工具 <a hreflang="en" href="https://www.figma.com/blog/webassembly-cut-figmas-load-time-by-3x/">Figma</a> 使用该技术显著提高了其性能。最近 <a hreflang="en" href="https://web.dev/ps-on-the-web/">Photoshop</a> 也出于类似的原因在使用 WebAssemply。
+
+## 研究方法
+
+WebAssembly 作为编译目标，以二进制模块分发。因此，我们在分析其在网络上的使用时面临着许多挑战。2021 Web 年鉴中的 WebAssembly 是其第一版，其中详细地介绍了[研究方法](../../en/2021/webassembly#methodology)和相关注意事项。2022 年的调研结果采用了相同的方法，唯一的增强是新增了一种用于确定编写 WebAssembly 模块语言的机制，分析的准确性会在各自章节中详细介绍。
+
+## WebAssembly 的使用范围有多广？
+
+我们在桌面端和移动端应用程序中分别找到了 3204 个和 2777 个确认的 WebAssembly 请求，它们分别在 2524 个和 2216 个域名中使用，分别占各自域名的 0.06% 和 0.04%。
+
+这表示我们在爬虫中发现模块的数量略有下降，桌面端和移动端请求分别减少了 16% 和 12%。但这并不一定意味着 WebAssembly 正在衰落，在解释这一变化时，以下几点值得注意：
+
+- 虽然你可以使用 WebAssembly 创建各种基于 Web 的内容，但它的主要优势是在复杂业务线具有大型代码库的应用程序中使用，这些应用通常已有多年历史（例如 Google Earth、Photoshop、AutoCAD）。这些 Web “应用程序” 并不像网站那么多，同时由于 Almanac 的爬虫网络主要是基于一些不太使用 WebAssembly 的站点首页组成，所以它们也并不总是可以被统计到。
+- 正如我们将在后面章节中看到的那样，大部分 WebAssembly 的使用都是来自于数量相对较少的第三方库。因此，这些库中的任何一个发生微小的变化都会对我们找到的模块数量产生重大影响。
+
+我们发现在移动端浏览器中使用 WebAssembly 模块更少一些（-13%），但这并不是移动端浏览器 WebAssembly 功能的反映，移动端浏览器通常具有更好的支持。相反，更可能是因为[渐进式增强](https://developer.mozilla.org/en-US/docs/Glossary/Progressive_Enhancement)标准的实践。在这些场景中，越是高级功能需要 WebAssembly 反而不支持。
```

Comment:
Would you mind checking if translations exist for MDN links? In this case it does - note I've not checked the rest.

```suggestion
我们发现在移动端浏览器中使用 WebAssembly 模块更少一些（-13%），但这并不是移动端浏览器 WebAssembly 功能的反映，移动端浏览器通常具有更好的支持。相反，更可能是因为[渐进式增强](https://developer.mozilla.org/zh-CN/docs/Glossary/Progressive_Enhancement)标准的实践。在这些场景中，越是高级功能需要 WebAssembly 反而不支持。
```

Reply:
yes of course. I'll check all the links.